src/plugins/data/common/opensearch_query/filters/geo_shape_filter.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { GeoShapeRelation } from '@opensearch-project/opensearch/api/types';
+import { Filter, FilterMeta } from './meta_filter';
+
+export type Position = number[];
+
+export interface PreIndexedShapeFilter {
+  index: string;
+  id: string;
+  path: string;
+  routing?: string;
+}
+
+export interface Polygon {
+  type: 'Polygon';
+  coordinates: Position[][];
+}
+
+export interface MultiPolygon {
+  type: 'MultiPolygon';
+  coordinates: Position[][][];
+}
+
+// TODO: support other geometries too.
+export type ShapeFilter = Polygon | MultiPolygon;
+
+export type GeoShapeFilterMeta = FilterMeta & {
+  params: {
+    shape?: ShapeFilter;
+    indexed_shape?: PreIndexedShapeFilter;
+    relation?: GeoShapeRelation;
+  };
+};
+
+export type GeoShapeFilter = Filter & {
+  meta: GeoShapeFilterMeta;
+  geo_shape: any;
```

Comment:
Will we consider define a more specific interface or type for the geo_shape property?

Reply:
geo_shape will have dynamic field name and "ignored_unmapped" as well. I couldn't find a way to define both in same structure. Do you know how to do that?

Reply:
Not quite sure about what should be geo_shape, but it is similar to meta, will this work?

```
export type GeoShapeFilter = Filter & {
  meta: GeoShapeFilterMeta;
  geo_shape: {
    [fieldName: string]: {
      shape?: ShapeFilter;
      indexed_shape?: PreIndexedShapeFilter;
      relation?: GeoShapeRelation;
      ignore_unmapped?: boolean;
    };
  };
};

```

Reply:
Not a blocker. I will just approve it.

Reply:
Here ignore_unmapped should be inside geo_shape , something like below
```
geo_shape: {
    [fieldName: string]: {
      shape?: ShapeFilter;
      indexed_shape?: PreIndexedShapeFilter;
      relation?: GeoShapeRelation;
    },
   ignore_unmapped?: boolean;
}
```
but typescript don't like this, since ignore_unmapped is also string and it can't differentiate whether is it part of fieldName or ignore_mapped.